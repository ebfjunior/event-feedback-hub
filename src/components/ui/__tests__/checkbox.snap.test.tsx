import React from 'react';
import { render } from '@testing-library/react';
import { Checkbox } from '../checkbox';

vi.mock('lucide-react', async () => {
	const actual = await vi.importActual<any>('lucide-react');
	return { ...actual, CheckIcon: (p: any) => <svg data-icon="check" {...p} /> };
});

describe('ui/Checkbox (snapshot)', () => {
	it('renders default', () => {
		const { container } = render(<Checkbox />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


