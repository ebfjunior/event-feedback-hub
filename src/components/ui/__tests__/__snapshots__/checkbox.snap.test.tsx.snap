// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/Checkbox (snapshot) > renders default 1`] = `
<button
  aria-checked="false"
  class="peer border-input dark:bg-input/30 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground dark:data-[state=checked]:bg-primary data-[state=checked]:border-primary focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50"
  data-slot="checkbox"
  data-state="unchecked"
  role="checkbox"
  type="button"
  value="on"
/>
`;
